docs/source/pagination/introduction.mdx

```diff
@@ -0,0 +1,127 @@
+
+## Subscribing to results
+
+The `GraphQLQueryPager` class can fetch data, but the caller must subscribe to the results in order to receive the data. A `subscribe` method is provided which takes a closure that is called whenever the pager fetches a new page of data. The `subscribe` method is a convenience method that ensures that the closure is called on the main thread.
+
+```swift
+// Guaranteed to run on the main thread
+pager.subscribe { result in
+    switch result {
+    case .success(let data):
+        // Handle the data
+    case .failure(let error):
+        // Handle the error
+    }
+}
+```
+
+The `GraphQLQueryPager` is also a Combine `Publisher`. As such, all `Publisher` methods are available, such as `sink`, `assign`, `map`, `filter`, and so on. 
+
+```swift
+// Can run on any thread
+pager.sink { result in
+    switch result {
+    case .success(let data):
+        // Handle the data
+    case .failure(let error):
+        // Handle the error
+    }
+}
+```
+
+## Custom response types
+
+The `GraphQLQueryPager` class is designed to work with any response type. By default, when initialized without a `transform`, the `GraphQLQueryPager` class will return a `Result` of `PaginationOutput`. This object contains three fields: `previousPages`, `initialPage`, and `nextPages`. 
+
+The `previousPages` field contains all the pages that have been fetched in the reverse direction, and exposes the paginated query's `Data` type. The `initialPage` field contains the first page of data that was fetched, and is an instance of the initial query's `Data` type. The `nextPages` field contains all the pages that have been fetched in the forward direction, and is a list of the paginated query's `Data` type.
+
+If you want to use a custom response type, you can provide a `transform` closure when initializing the `GraphQLQueryPager` instance. This closure is called whenever the pager fetches a new page of data, and is used to transform the raw response into the desired response type. 
+
+```swift
+struct Person {
+    let id: String
+    let name: String
+}
+
+let pager = GraphQLQueryPager.makeForwardCursorQueryPager(
+    client: apolloClient,
+    queryProvider: { page in
+        MyQuery(first: 10, after: page?.cursor ?? .none)
+    },
+    extractPageInfo: { data in
+        CursorBasedPagination.Forward(
+            hasNext: data.values.pageInfo.hasNextPage ?? false,
+            endCursor: data.values.pageInfo.endCursor
+        )
+    },
+    transform: { data in 
+        data.values.nodes.map { Person(id: $0.id, name: $0.name) }
+    }
+)
+```
+
+In this example, the `GraphQLQueryPager` instance is initialized with a `transform` closure that transforms the raw response into an array of `Person` objects. Alternatively, we can make use of the `GraphQLQueryPager`'s `map` method to achieve the same result:
```

Comment:
I think we need to make it clear that this `transform` will be called on the `data` for each page query's response sequentially and then the arrays will be concatenated together.

And alternatively, you can use the function with the `transform` signature `([PaginatedQuery.Data], InitialQuery.Data, [PaginatedQuery.Data]) throws -> Model` to get all of the page data at once and manually build whatever data structure you want. (With a quick example of what this looks like as well.)